codesign/lib/src/utils.dart

```diff
@@ -5,10 +5,45 @@
 import 'dart:io';
 
 import 'package:args/args.dart';
+import 'package:file/file.dart';
 import 'package:process/process.dart';
 
 import 'log.dart';
 
+const String gsCloudBaseUrl = r'gs://flutter_infra_release';
+
+Future<void> upload({
+  required String localPath,
+  required String remotePath,
+  required String commitHash,
+  required ProcessManager processManager,
+}) async {
+  final String fullRemotePath = '$gsCloudBaseUrl/flutter/$commitHash/$remotePath';
+  final ProcessResult result = await processManager.run(
+    <String>['gsutil', 'cp', localPath, fullRemotePath],
+  );
+  if (result.exitCode != 0) {
+    throw Exception('Failed to upload $localPath to $fullRemotePath');
+  }
+}
+
+Future<File> download({
+  required String remotePath,
+  required String localPath,
+  required String commitHash,
```

Comment:
Consider adding a wrapper function, like `downloadEngineArtifacts()` as a generic `download` function will only have a `uri` parameter and the `destination` of the file

Reply:
Sure, added wrapper for both download and upload.